### PR TITLE
feat!: Migrate Okta plugin to v2

### PR DESCRIPTION
**Pull request by disq**

### CODE REVIEW

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Can you please open a new okta account and manually test it ?

**Comment by disq** [comment]
> Can you please open a new okta account and manually test it ?

seems to work!

File: plugins/source/okta/client/spec.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package client
+
+type Spec struct {
+	Token  string `yaml:"token,omitempty"`
```
Comment (yevgenypats) [inline plugins/source/okta/client/spec.go:4]: it should be json (we use json everywhere apart from the first step in the CLI where we read yaml and then convert to json - just like in k8s)

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Looks good. one comment and ready to go.

**Review by yevgenypats — APPROVED** [review]

**Pull request merged**